satpy/readers/seviri_base.py

```diff
@@ -372,14 +377,92 @@ def _tl15(self, data, wavenumber):
         return ((C2 * wavenumber) /
                 np.log((1.0 / data) * C1 * wavenumber ** 3 + 1.0))
 
-    def _vis_calibrate(self, data, solar_irradiance):
+    def vis_calibrate(self, data, solar_irradiance):
         """Calibrate to reflectance.
 
         This uses the method described in Conversion from radiances to
         reflectances for SEVIRI warm channels: https://tinyurl.com/y67zhphm
         """
         reflectance = np.pi * data * 100.0 / solar_irradiance
-        return apply_earthsun_distance_correction(reflectance, self.start_time)
+        return apply_earthsun_distance_correction(reflectance, self.scan_time)
+
+
+class SEVIRICalibrationHandler:
+    """Calibration handler for SEVIRI HRIT-, native- and netCDF-formats.
+
+    Handles selection of calibration coefficients and calls the appropriate
+    calibration algorithm.
+    """
+
+    def __init__(self, platform_id, channel_name, coefs, calib_mode, scan_time):
+        """Initialize the calibration handler."""
+        self.platform_id = platform_id
+        self.channel_name = channel_name
+        self.coefs = coefs
+        self.calib_mode = calib_mode.upper()
+        self.scan_time = scan_time
+        self.algo = SEVIRICalibrationAlgorithm(
+            platform_id=self.platform_id,
+            scan_time=self.scan_time
+        )
```

Comment:
Do all these attributes need to be public?

Reply:
No, I think they can be private